dimensiondata/src/main/java/org/jclouds/dimensiondata/cloudcontrol/features/NetworkApi.java

```diff
@@ -0,0 +1,453 @@
+   Response deployVlan(@PayloadParam("networkDomainId") String networkDomainId, @PayloadParam("name") String name,
+         @PayloadParam("description") String description,
+         @PayloadParam("privateIpv4BaseAddress") String privateIpv4BaseAddress,
+         @PayloadParam("privateIpv4PrefixSize") Integer privateIpv4PrefixSize);
+
+   @Named("network:getVlan")
+   @GET
+   @Path("/vlan/{id}")
+   @Fallback(NullOnNotFoundOr404.class)
+   Vlan getVlan(@PathParam("id") String vlanId);
+
+   @Named("network:vlan")
+   @GET
+   @Path("/vlan")
+   @ResponseParser(ParseVlans.class)
+   @Fallback(Fallbacks.EmptyIterableWithMarkerOnNotFoundOr404.class)
+   PaginatedCollection<Vlan> listVlans(@QueryParam("networkDomainId") String networkDomainId,
+         PaginationOptions options);
+
+   @Named("network:vlan")
+   @GET
+   @Path("/vlan")
+   @Transform(ParseVlans.ToPagedIterable.class)
+   @ResponseParser(ParseVlans.class)
+   @Fallback(Fallbacks.EmptyPagedIterableOnNotFoundOr404.class)
+   PagedIterable<Vlan> listVlans(@QueryParam("networkDomainId") String networkDomainId);
+
+   @Named("network:listVlansWithNetworkDomainIdAndName")
+   @GET
+   @Path("/vlan")
+   @Transform(ParseVlans.ToPagedIterable.class)
+   @ResponseParser(ParseVlans.class)
+   @Fallback(Fallbacks.EmptyPagedIterableOnNotFoundOr404.class)
+   PagedIterable<Vlan> listVlansWithNetworkDomainIdAndName(@QueryParam("networkDomainId") String networkDomainId,
+         @QueryParam("name") String name);
+
+   @Named("network:deleteVlan")
+   @POST
+   @Path("/deleteVlan")
+   @Produces(MediaType.APPLICATION_JSON)
+   @MapBinder(BindToJsonPayload.class)
+   @Fallback(NullOnNotFoundOr404.class)
+   Response deleteVlan(@PayloadParam("id") String vlanId);
+
+   @Named("network:addPublicIpBlock")
+   @POST
+   @Path("/addPublicIpBlock")
+   @Produces(MediaType.APPLICATION_JSON)
+   @MapBinder(BindToJsonPayload.class)
+   Response addPublicIpBlock(@PayloadParam("networkDomainId") String networkDomainId);
+
+   @Named("network:publicIpBlock")
+   @GET
+   @Path("/publicIpBlock")
+   @ResponseParser(ParsePublicIpBlocks.class)
+   @Fallback(Fallbacks.EmptyIterableWithMarkerOnNotFoundOr404.class)
+   PaginatedCollection<PublicIpBlock> listPublicIPv4AddressBlocks(@QueryParam("networkDomainId") String networkDomainId,
+         PaginationOptions options);
+
+   @Named("network:publicIpBlock")
+   @GET
+   @Path("/publicIpBlock")
+   @Transform(ParsePublicIpBlocks.ToPagedIterable.class)
+   @ResponseParser(ParsePublicIpBlocks.class)
+   @Fallback(Fallbacks.EmptyPagedIterableOnNotFoundOr404.class)
+   PagedIterable<PublicIpBlock> listPublicIPv4AddressBlocks(@QueryParam("networkDomainId") String networkDomainId);
+
+   @Named("network:removePublicIpBlock")
+   @POST
+   @Path("/removePublicIpBlock")
+   @Produces(MediaType.APPLICATION_JSON)
+   @MapBinder(BindToJsonPayload.class)
+   @Fallback(NullOnNotFoundOr404.class)
+   Response removePublicIpBlock(@PayloadParam("id") String publicIpBlockId);
```

Comment:
jclouds uses @Fallback(VoidOnNotFoundOr404.class) and uses void delete*(...)
do you think Response is more useful here?